game/tests.py

```
@@ -3604,126 +3492,104 @@ def test_upload_with_no_file_shows_error(self):
+        self.assertEqual(calculate_rating_change("win", 1200, 800), 3)
+        # player=1200, opponent=800, loss
+        self.assertEqual(calculate_rating_change("loss", 1200, 800), -29)
+        # player=1200, opponent=1200, win
+        self.assertEqual(calculate_rating_change("win", 1200, 1200), 16)
+        # player=1200, opponent=1600, win
+        self.assertEqual(calculate_rating_change("win", 1200, 1600), 29)
+        # player=1200, opponent=1600, loss
+        self.assertEqual(calculate_rating_change("loss", 1200, 1600), -3)
+
+        # Invalid result raises ValueError
+        with self.assertRaises(ValueError):
```

Ruff (0.15.20)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)